internal/humanitec/convert.go

```diff
@@ -84,8 +86,70 @@ func getProbeDetails(probe *score.ContainerProbeSpec) map[string]interface{} {
 	return res
 }
 
+// readFile reads a text file into memory
+func readFile(baseDir, path string) (string, error) {
+	if !filepath.IsAbs(path) {
+		path = filepath.Join(baseDir, path)
+	}
+
+	raw, err := os.ReadFile(path)
+	if err != nil {
+		return "", fmt.Errorf("reading '%s': %w", path, err)
+	}
+
+	return string(raw), nil
+}
+
+// mergeFileContent joins inline file contents with '\n' character (DEPRECATED)
+func mergeFileContent(content interface{}, target string) (string, error) {
+	switch val := content.(type) {
+	case string:
+		return val, nil
+	case []interface{}:
+		// TODO: Deprecated functionality
+		log.Printf("Warning: The content for the '%s' file is provided in a deprecated format. Strings will be joined with '\\n' character.", target)
+		var sb strings.Builder
+		for _, str := range val {
+			if sb.Len() > 0 {
+				sb.WriteString("\n")
+			}
+			sb.WriteString(fmt.Sprint(str))
+		}
+		return sb.String(), nil
+		// END (TODO)
+	default:
+		return "", fmt.Errorf("can not use '%T' as a content for '%s': not supported", content, target)
+	}
+}
+
+// convertFileMountSpec extracts a mount file detailf from the source spec.
+func convertFileMountSpec(f *score.FileMountSpec, context *templatesContext, baseDir string) (string, map[string]interface{}, error) {
+	var err error
+	var content string
+
+	if f.Source != "" {
+		content, err = readFile(baseDir, f.Source)
+	} else {
+		content, err = mergeFileContent(f.Content, f.Target)
+	}
+	if err != nil {
+		return "", nil, err
+	}
+
+	if !f.NoExpand {
```

Comment:
We actually need to do more here. Placeholders themselves need to be escaped if `NoExpand` is true.

E.g. with `NoExpand = true` the following should happen:

`${do-not-interpret}` -> `$\{do-not-intrepret}` As described here: https://developer.humanitec.com/platform-orchestrator/reference/placeholders

Reply:
I would disagree. The `noExpand` switch controls `score-humanitec` CLI tool behavior. It should not modify the source file.

If the User wants to control how Humanitec passes through the values, they should escape those in Score file explicitly. E.g. if the User would want to achieve this by editing values in Humanitec web UI, what value would they put into the edit box? Same value they should use in Score file.

Reply:
That is not the way to think about it. The idea here is that the user sepciofies in the score file that *no escaping happens* In general they do not know what score implementation will be used. From the view of the user score-humanitec = Humanitec so the behaviour in teh score file is the outcome they are looking for not the input to Humanitec.